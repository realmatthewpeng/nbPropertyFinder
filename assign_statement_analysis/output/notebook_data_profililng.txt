Total number of lines found: 25

data = pandas.read_csv(data_path)
describe_transposed = describe.T
describe = data.describe(include = "all")
describe_final = describe_transposed.reset_index()
dtypes = data.dtypes
dtypes_dataframe = pandas.DataFrame(dtypes, columns = ["data_type"])
dtypes_final = dtypes_dataframe.reset_index()
null_count_series = data.isnull().sum().astype(int)
filled_count_series = data.notnull().sum().astype(int)
percent_null =  null_count_series / totalcount
percent_filled = filled_count_series / totalcount
percent_null = percent_null.reset_index().rename(columns = {0:"null_percent"})
null_count_series = null_count_series.reset_index().rename(columns = {0:"null_counts"})
percent_filled = percent_filled.reset_index().rename(columns = {0:"non_null_percent"})
filled_count_series = filled_count_series.reset_index().rename(columns = {0:"non_null_counts"})
null_info_dataframe = reduce(lambda left,right: pandas.merge(left,right,on='index'), [percent_null,percent_filled,filled_count_series,null_count_series])
merged = reduce(lambda left, right: pandas.merge(left, right, on='index', how="left"),[null_info_dataframe, dtypes_final, describe_final])
merged["row_count"] = totalcount
merged["data_type"] = merged["data_type"].astype(str)
merged[column] = merged[column].astype(float).round(rounded_places)
data_no_nulls = data.dropna(how = "any", axis = 0)
data_no_nulls[column] = data_no_nulls[column].fillna(data_no_nulls[column].mean())
data_no_nulls = data
data_no_nulls[column] = data_no_nulls[column].astype(str)
correlation_dataframe = data_no_nulls.corr()

Incorrect Lines:

percent_null =  null_count_series / totalcount
percent_filled = filled_count_series / totalcount

Precision = 0.92
Recall = 0.9583333333333334

dtype = data_no_nulls[column].dtype
