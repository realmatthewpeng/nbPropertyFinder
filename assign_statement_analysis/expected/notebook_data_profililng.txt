describe = data.describe(include = "all")
describe_final = describe_transposed.reset_index()
dtypes = data.dtypes
dtypes_final = dtypes_dataframe.reset_index()
filled_count_series = data.notnull().sum().astype(int)
null_count_series = data.isnull().sum().astype(int)
percent_null = percent_null.reset_index().rename(columns = {0:"null_percent"})
percent_filled = percent_filled.reset_index().rename(columns = {0:"non_null_percent"})
filled_count_series = filled_count_series.reset_index().rename(columns = {0:"non_null_counts"})
null_count_series = null_count_series.reset_index().rename(columns = {0:"null_counts"})
null_info_dataframe = reduce(lambda left,right: pandas.merge(left,right,on='index'), [percent_null,percent_filled,filled_count_series,null_count_series])
merged = reduce(lambda left, right: pandas.merge(left, right, on='index', how="left"),[null_info_dataframe, dtypes_final, describe_final])
merged["row_count"] = totalcount
merged["data_type"] = merged["data_type"].astype(str)
merged[column] = merged[column].astype(float).round(rounded_places)
data_no_nulls = data.dropna(how = "any", axis = 0)
data_no_nulls[column] = data_no_nulls[column].fillna(data_no_nulls[column].mean())
dtype = data_no_nulls[column].dtype
data_no_nulls[column] = data_no_nulls[column].astype(str)
data = pandas.read_csv(data_path)
describe_transposed = describe.T
dtypes_dataframe = pandas.DataFrame(dtypes, columns = ["data_type"])
data_no_nulls = data
correlation_dataframe = data_no_nulls.corr()